implementations/rust/ockam/ockam_node/src/storage/database/migrations/migration_support/migrator.rs

```diff
@@ -266,19 +185,192 @@ impl Migrator {
 
     /// Run all migrations
     pub async fn migrate(&self, pool: &SqlitePool) -> Result<()> {
-        self.migrate_partial(pool, 0, i64::MAX, false).await
+        self.migrate_up_to(pool, i64::MAX).await
     }
 }
 
 #[cfg(test)]
 impl Migrator {
-    /// Migrate the schema of the database right before the specified version
-    pub(crate) async fn migrate_before(
-        &self,
+    /// Run migrations up to the specified version (inclusive) but skip the last rust migration
+    pub(crate) async fn migrate_up_to_skip_last_rust_migration(
+        mut self,
         pool: &SqlitePool,
-        version: i64, // not including
-        run_last_sql: bool,
+        up_to: Version,
+    ) -> Result<()> {
+        self.rust_migrations.retain(|m| m.version() < up_to);
```

Comment:
Would be enough to just remove one rust migration where `m.version() == up_to` so that you don't repeat the filtering logic twice